src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/FileSystem.Portable.vb

```diff
@@ -0,0 +1,1522 @@
+#End If
+                Exit Sub
+            End If
+
+            ' Use Framework.
+            If operation = CopyOrMove.Copy OrElse
+                sourceFileFullPath.Equals(destinationFileFullPath, StringComparison.OrdinalIgnoreCase) Then
+                ' Call IO.File.Copy if this is a copy operation.
+                ' In addition, if sourceFileFullPath is the same as destinationFileFullPath,
+                '   IO.File.Copy will throw, IO.File.Move will not (VSWhidbey334759).
+                '   Whatever overwrite flag is passed in, IO.File.Move should throw exception,
+                '   so call IO.File.Copy to get the exception as well.
+                IO.File.Copy(sourceFileFullPath, destinationFileFullPath, overwrite)
+            Else ' MoveFile with support for overwrite flag.
+                If overwrite Then ' User wants to overwrite destination.
+                    ' Why not checking for destination existence: user may not have read permission / ACL,
+                    ' but have write permission / ACL thus cannot see but can delete / overwrite destination.
+
+                    'If Environment.OSVersion.Platform = PlatformID.Win32NT Then ' Platforms supporting MoveFileEx.
+                    '    ' SECURITY NOTE: We already demand permission at the start.
+                    '    'Call (New SecurityPermission(SecurityPermissionFlag.UnmanagedCode)).Assert()
+                    '    Try
+                    '        IO.File.Move(sourceFileName:=sourceFileFullPath, destFileName:=destinationFileFullPath)
+                    '        ' GetLastWin32Error has to be close to PInvoke call. FxCop rule.
+                    '        If Not succeed Then
+                    '            ThrowWinIOError(System.Runtime.InteropServices.Marshal.GetLastWin32Error())
+                    '        End If
+                    '    Catch
+                    '        Throw
+                    '    Finally
+                    '    End Try
+                    'Else ' Win95, Win98, WinME.
+                    ' IO.File.Delete will not throw if destinationFileFullPath does not exist
+                    ' (user may not have permission to discover this, but have permission to overwrite),
+                    ' so always delete the destination.
+                    IO.File.Delete(destinationFileFullPath)
+                    IO.File.Move(sourceFileFullPath, destinationFileFullPath)
+                    'End If
+                Else ' Overwrite = False, call Framework.
+                    IO.File.Move(sourceFileFullPath, destinationFileFullPath)
+                End If ' Overwrite
+            End If
+        End Sub
+
+        Private Shared Sub DeleteDirectoryInternal(ByVal directory As String, ByVal onDirectoryNotEmpty As DeleteDirectoryOption,
+                    ByVal showUI As UIOptionInternal, ByVal recycle As RecycleOption, ByVal onUserCancel As UICancelOption)
+
+            ' Verify enums. VSWhidbey 522083.
```

Comment:
Anything like VSWhidbey nnnnnn is a bug number (nearly 15 years old in this case) - please delete all bug numbers, we will not need them.

Reply:
There are other references to VSWhidbey that are not in my code so I left them